docs/design/email_notification.md

```diff
@@ -0,0 +1,47 @@
+Email
+---
+
+#### Considerations
+
+- Be able to send frontpage email notifications on a daily or weekly basis
+  - Frequency is user-configurable
+  - If we don't have any new content since the last send, defer until next interval
+
+
+- Be able to notify users on specific events, controlled by a subscription to the entity
+  - Examples:
+    - Comment replies to posts
+    - Comment replies to other comments
+  - Capability to notify at the following frequencies:
+    - Near term we want to support:
+      - Never
+      - Immediate
+    - Long term we may want to support:
+      - Daily (digest/aggregation of new events)
+      - Weekly (digest/aggregation of new events)
+
+
+#### Proposal
+
+For frontpage digests the workflow would be:
+
+ - Daily and weekly tasks to:
+  - Walk `NotificationSettings` matching the corresponding `notification_type`
+  - For each of these settings, get frontpage data for that user
+  - Filter the list of frontpage posts to those that were made since the last frontpage email we sent this user
+    - A record needs to be stored at send time of when that happened
+  - If we have at least 1 post to send, render and send the emails
```

Comment:
If the user turns off the daily notification, doesn't use OD for three months, and then comes back and turns it back on, do they receive an email with all posts from the last three months? If not, do we just have an upper limit (the email always includes 10 posts max, for instance) or do we treat turning the notification back on as the timestamp we use for cutting off posts for the new email?

Reply:
The tickets for [daily](https://github.com/mitodl/open-discussions/issues/461) and [weekly](https://github.com/mitodl/open-discussions/issues/460) have a requirement for it to be the top `n` posts. So I would expect turning it back on to have it send the top n posts since the last email we sent the user (could be never, days, months, years, etc ago). But for the frontpage, those items are computed at send-time because it's a snapshot of the frontpage at that particular moment, it's not an aggregation of discrete events like comment notifications.

Reply:
Cool, that makes sense to me then